crates/agglayer-certificate-orchestrator/src/certificate_task.rs

```diff
@@ -520,6 +539,10 @@ where
             %settlement_tx_hash,
             "Waiting for certificate settlement to complete"
         );
+
+        // Associate the pp root with the certificate ID.
+        self.state_store
+            .add_certificate_id_for_pp_root(&PessimisticRoot::from(new_pp_root), &certificate_id)?;
```

Comment:
Same question - is this one needed? `self.new_pp_root` is only changed in `process_from_pending`.